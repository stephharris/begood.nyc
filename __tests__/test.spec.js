import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Home from '../views/react/listings';
import Listings from '../views/react/listings';
import ListingContracted from '../views/react/listing-contracted';
import ListingExpanded from '../views/react/listing-expanded';

test('Should render 0 Listings as expected', () => {
  const listings = [];
  const component = shallow(<Listings listings={listings}/>)
  const tree = toJson(component)
  expect(component.find(ListingContracted).length).toBe(0);
  expect(component.contains('awaiting more opportunities')).toBe(true)
  expect(tree).toMatchSnapshot();
})

test('Should properly render multiple contracted listings components', () => {
  let opportunity = '';
  const listings = [{
    active: true,
    title: 'Test Title',
    timeCommitment: 'ongoing',
    hours: 'scheduling tbd',
    neighborhood: 'Downtown Brooklyn',
    borough: 'Brooklyn',
    briefDescription: 'testing testing 123'
  },{
    active: true,
    title: 'Test Title 2',
    timeCommitment: 'ongoing',
    hours: 'scheduling tbd',
    neighborhood: 'Downtown Brooklyn',
    borough: 'Brooklyn',
    briefDescription: 'testing testing 123'
  }]
  const component = shallow(<Listings opportunity={opportunity} listings={listings}/>)
  const tree = toJson(component)
  expect(component.find(ListingContracted).length).toBe(2)
  expect(tree).toMatchSnapshot();
})

test('Should properly render singular expanded listings component', () => {
  let opportunity = '#Test_Title';
  let testRoute = 'Test_Title';
  let date = new Date();
  const listings = [{
    active: true,
    title: 'Test Title',
    timeCommitment: 'ongoing',
    hours: 'scheduling tbd',
    neighborhood: 'Downtown Brooklyn',
    borough: 'Brooklyn',
    briefDescription: 'testing testing 123',
    route: 'Test_Title'
  }]
  const component = shallow(<Listings testRoute={testRoute} opportunity={opportunity} listings={listings}/>)
  const tree = toJson(component)
  expect(component.find(ListingExpanded).length).toBe(1)
  expect(tree).toMatchSnapshot();
})





