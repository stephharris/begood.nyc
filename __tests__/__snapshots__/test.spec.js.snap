// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should properly render multiple contracted listings components 1`] = `
<div>
  <div
    onClick={[Function]}
  >
    <ListingContracted
      listing={
        Object {
          "active": true,
          "borough": "Brooklyn",
          "briefDescription": "testing testing 123",
          "hours": "scheduling tbd",
          "neighborhood": "Downtown Brooklyn",
          "timeCommitment": "ongoing",
          "title": "Test Title",
        }
      }
    />
  </div>
  <div
    onClick={[Function]}
  >
    <ListingContracted
      listing={
        Object {
          "active": true,
          "borough": "Brooklyn",
          "briefDescription": "testing testing 123",
          "hours": "scheduling tbd",
          "neighborhood": "Downtown Brooklyn",
          "timeCommitment": "ongoing",
          "title": "Test Title 2",
        }
      }
    />
  </div>
</div>
`;

exports[`Should properly render singular expanded listings component 1`] = `
<div>
  <div>
    <ListingExpanded
      listing={
        Object {
          "active": true,
          "borough": "Brooklyn",
          "briefDescription": "testing testing 123",
          "hours": "scheduling tbd",
          "neighborhood": "Downtown Brooklyn",
          "route": "Test_Title",
          "timeCommitment": "ongoing",
          "title": "Test Title",
        }
      }
      toggleView={[Function]}
    />
  </div>
</div>
`;

exports[`Should render 0 Listings as expected 1`] = `
<div>
  <h3>
    awaiting more opportunities
  </h3>
</div>
`;
